sycl/source/detail/platform_util.cpp

```diff
@@ -62,21 +72,49 @@ uint32_t PlatformUtil::getMaxClockFrequency() {
   Buff = Buff.substr(Buff.rfind(' '), Buff.length());
   Freq *= std::stod(Buff);
   return Freq;
+#else
+#warning Your platform is not supported!
+#endif
+  return 0;
 }
 
 uint32_t PlatformUtil::getMemCacheLineSize() {
+#if defined(__x86_64__) || defined(__i386__)
   uint32_t CPUInfo[4];
   cpuid(CPUInfo, 0x80000006);
   return CPUInfo[2] & 0xff;
+#elif defined(SYCL_RT_OS_LINUX) && defined(_SC_LEVEL2_DCACHE_LINESIZE)
+  long lineSize = sysconf(_SC_LEVEL2_DCACHE_LINESIZE);
+  if (lineSize > 0) {
+    return lineSize;
+  }
+#else
+#warning Your platform is not supported.
```

Comment:
#warning is not portable directive.
On Windows I see a builtbot fail "invalid preprocessor command 'warning'".
MSVC requires something like #pragma message("abc")

Reply:
Which version doesn't support #pragma warning?  Newer versions do at least.

Reply:
I should just remove the warning.  I'm not sure how useful it is to have this.  Documentation of what is and is not supported is better.

Reply:
Implemented in 95c3fb515aa3